rclpy/rclpy/executors.py

```diff
@@ -756,7 +805,7 @@ def _wait_for_ready_callbacks(
 
                 for gc in node.guards:
                     if gc._executor_triggered:
-                        if gc.callback_group.can_execute(gc):
+                        if gc.callback_group and gc.callback_group.can_execute(gc):
```

Comment:
Why do we not need to add the same check for waitables, subscriptions, clients and services?

Reply:
Since `GuardCondition` can have Optional `.callback_group`. And the others always have a `CallbackGroup`